rest/rest-server/src/test/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRestPaginationTest.java

```diff
@@ -0,0 +1,166 @@
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ * ################################################################
+ * $$PROACTIVE_INITIAL_DEV$$
+ */
+package org.ow2.proactive_grid_cloud_portal.scheduler;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobState;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.TaskState;
+import org.ow2.proactive.scheduler.common.util.SchedulerProxyUserInterface;
+import org.ow2.proactive_grid_cloud_portal.RestTestServer;
+import org.ow2.proactive_grid_cloud_portal.common.SchedulerRestInterface;
+import org.ow2.proactive_grid_cloud_portal.common.SharedSessionStoreTestUtils;
+import org.ow2.proactive_grid_cloud_portal.scheduler.dto.TaskStateData;
+
+
+/**
+ * The [offset, limit[ boundaries and tag filtering are out of the scope of these tests
+ * because their control is done in the <code>JobState</code> class.
+ * We are only testing the REST interface methods providing the pagination on tasks.
+ * @author paraita
+ *
+ */
+public class SchedulerStateRestPaginationTest extends RestTestServer {
+
+    private SchedulerRestInterface restInterface = null;
+    private SchedulerProxyUserInterface mockOfScheduler = null;
+    private String sessionId = null;
+
+    @Before
+    public void setUp() throws Throwable {
+        restInterface = new SchedulerStateRest();
+        mockOfScheduler = mock(SchedulerProxyUserInterface.class);
+        sessionId = SharedSessionStoreTestUtils.createValidSession(mockOfScheduler);
+    }
+
+    @Test
+    public void testGetJobTasksIdsPaginated() throws Throwable {
+
+        int nbTasks = 50;
+        String jobIdStr = "1";
+        JobState job = newMockedJob(jobIdStr, nbTasks);
+        when(mockOfScheduler.getJobState(jobIdStr)).thenReturn(job);
+
+        List<String> res = restInterface.getJobTasksIdsPaginated(sessionId, jobIdStr, 0, nbTasks);
+
+        assertEquals("Number of tasks is incorrect", nbTasks, res.size());
+
+        for (int i = 0; i < nbTasks; i++) {
+            assertEquals("Task readable name is incorrect", generateReadableName(jobIdStr, i, nbTasks),
+                    res.get(i));
+        }
+
+    }
+
+    @Test
+    public void testGetJobTasksIdsByTagPaginated() throws Throwable {
+        int nbTasks = 50;
+        String jobIdStr = "1";
```

Comment:
The three next lines could be put in a new method two factorize code in lot of test methods.